wsmaster/che-core-api-workspace-activity/src/main/java/org/eclipse/che/api/workspace/activity/WorkspaceActivityChecker.java

```diff
@@ -95,6 +96,12 @@ void validate() {
     } catch (ServerException e) {
       LOG.error(e.getLocalizedMessage(), e);
     }
+
+    try {
+      reconcileActivityStatuses();
```

Comment:
Just a minor assumption - since it should be more exception than the rule(I mean outdated activity statuses) maybe we should do it less frequently, like introduce new delay parameter or do it like every 1+n time(like 1, 5, 9, ...) when `validate` is invoked.

Reply:
I've added the new config parameter with the default period of 1 hour.